apps/coscrad-frontend/src/components/resources/vocabulary-lists/vocabulary-list-detail.full-view.presenter.tsx

```diff
@@ -1,21 +1,95 @@
+
+type VocabularyListFilterProperty = string | boolean;
+
+export type VocabularyListFilter = Record<string, VocabularyListFilterProperty>;
+
+const UPDATE_VOCABULARY_LIST_FILTER = 'UPDATE_VOCABULARY_LIST_FILTER';
+
+type UpdateVocabularyListFilterPayload = {
+    key: string;
+    value: string | boolean;
+};
+
+type FSA<T, U> = {
+    type: T;
+    payload: U;
+};
+
+const updateVocabularyListFilter = (
+    key: string,
+    value: VocabularyListFilterProperty
+): FSA<typeof UPDATE_VOCABULARY_LIST_FILTER, UpdateVocabularyListFilterPayload> => ({
+    type: UPDATE_VOCABULARY_LIST_FILTER,
+    payload: {
+        key,
+        value,
+    },
+});
+
+const filterReducer = (
+    state: VocabularyListFilter,
+    {
+        type,
+        payload: { key, value },
+    }: FSA<typeof UPDATE_VOCABULARY_LIST_FILTER, UpdateVocabularyListFilterPayload>
+) => {
+    if (type !== UPDATE_VOCABULARY_LIST_FILTER) return state;
+
+    return {
+        /**
+         * Note that we **are** immutably updating state since each property
+         * is a primitive (`string` | `boolean`) and hence a shallow clone
+         * is as good as deep.
+         */
+        ...state,
+        [key]: value,
+    };
+};
+
+const filterEntriesForSelectedTerms = (
+    allEntries: IVocabularyListEntry<VocabularyListFilterProperty>[],
+    filter: VocabularyListFilter
+): IVocabularyListEntry<VocabularyListFilterProperty>[] =>
+    allEntries.filter(({ variableValues }) => doValuesMatchFilters(variableValues, filter));
 
 export const VocabularyListDetailFullViewPresenter = ({
-    data: { id, name, nameEnglish, entries },
-}: IDetailQueryResult<IVocabularyListViewModel>): JSX.Element => (
-    <div data-testid={id}>
-        <h1>{formatBilingualText(name, nameEnglish)}</h1>
-        <div>
-            <h3>Entries:</h3>
-            {/* TODO [https://www.pivotaltracker.com/story/show/183681556] We need to fix the following hack */}
-            <Carousel
-                propsForItems={entries.map(({ term }) => ({ data: term, actions: [] }))}
-                Presenter={TermDetailFullViewPresenter}
-            />
+    data: { id, name, nameEnglish, entries, form },
+}: IDetailQueryResult<IVocabularyListViewModel>): JSX.Element => {
+    const [filter, dispatch] = useReducer(filterReducer, {});
+
+    const selectedEntries = filterEntriesForSelectedTerms(entries, filter);
+
+    return (
+        <div data-testid={id}>
+            <h1>{formatBilingualText(name, nameEnglish)}</h1>
+            <div>
+                <h3>Entries:</h3>
+                {/* TODO [https://www.pivotaltracker.com/story/show/183681556] We need to fix the following hack */}
+                <Carousel
+                    propsForItems={selectedEntries.map(({ term }) => ({ data: term, actions: [] }))}
+                    Presenter={TermDetailFullViewPresenter}
+                />
+            </div>
+            <div>
+                <VocabularyListForm
```

Comment:
Does this belong in the presenter? Maybe the container should have the state and form and we should inject the rest of the presentation into the container.